packages/@ember/-internals/container/lib/registry.ts

```diff
@@ -396,7 +396,7 @@ export default class Registry implements IRegistry {
     this._typeOptions[type] = options;
   }
 
-  getOptionsForType(type: string): TypeOptions {
+  getOptionsForType(type: string): TypeOptions | undefined {
```

Comment:
If the type wasn't registered and there was no fallback, this would be undefined.